cmd/podman/images/scp.go

```diff
@@ -0,0 +1,344 @@
+		} else {
+			err := validateImageName(args[0])
+			if err != nil {
+				return nil, err
+			}
+			scpOpts.SourceImageName = args[0]
+		}
+	case 2:
+		if strings.Contains(args[0], "::") {
+			if !(strings.Contains(args[1], "::")) && len(strings.Split(args[0], "::")[1]) <= 1 { // if an image is specified, this mean we are loading from our client
+				cliConnections = append(cliConnections, args[0])
+				scpOpts.ToRemote = true
+			} else if strings.Contains(args[1], "::") { // both remote clients
+				scpOpts.FromRemote = true
+				scpOpts.ToRemote = true
+				if len(strings.Split(args[0], "::")[1]) <= 1 { // is save->load w/ one image name or save-load w/ two image names
+					cliConnections = append(cliConnections, args[0])
+					cliConnections = append(cliConnections, args[1])
+				} else if len(strings.Split(args[0], "::")[1]) > 1 && len(strings.Split(args[1], "::")[0]) > 1 {
+					return nil, errors.Wrapf(define.ErrInvalidArg, "cannot specify an image rename")
+				} else { // else its a load save (order of args)
+					cliConnections = append(cliConnections, args[1])
+					cliConnections = append(cliConnections, args[0])
+				}
+			} else {
+				return nil, errors.Wrapf(define.ErrInvalidArg, "cannot specify an image rename")
+			}
+		} else if strings.Contains(args[1], "::") { // if we are given image host::
+			if len((strings.Split(args[1], "::")[1])) > 1 {
+				return nil, errors.Wrapf(define.ErrInvalidArg, "cannot specify an image rename")
+			}
+			err := validateImageName(args[0])
+			if err != nil {
+				return nil, err
+			}
+			scpOpts.SourceImageName = args[0]
+			scpOpts.ToRemote = true
+			cliConnections = append(cliConnections, args[1])
+		} else {
+			return nil, errors.Wrapf(define.ErrInvalidArg, "cannot specify an image rename")
+		}
+	}
+	for i, val := range cliConnections {
+		splitEnv := strings.SplitN(val, "::", 2)
+		scpOpts.Connections = append(scpOpts.Connections, splitEnv[0])
+		if len(splitEnv[1]) != 0 {
+			err := validateImageName(splitEnv[1])
+			if err != nil {
+				return nil, err
+			}
+			scpOpts.SourceImageName = splitEnv[1]
+			//TODO: actually use the new name given by the user
+		}
+		conn, found := cfg.Engine.ServiceDestinations[scpOpts.Connections[i]]
+		if found {
+			iden = append(iden, conn.Identity)
+			url = append(url, conn.URI)
+		} else { // no match, warn user and do a manual connection.
+			url = append(url, "ssh://"+scpOpts.Connections[i])
+			iden = append(iden, "")
+			logrus.Warnf("Unknown connection name given. Please use system connection add to specify the default remote socket location")
+		}
+	}
+	for i, urlParse := range url {
+		urlt, err := urlP.Parse(urlParse) // create an actual url to pass to exec command
+		if err != nil {
+			return nil, err
+		}
+		scpOpts.URI = append(scpOpts.URI, urlt)
+		if scpOpts.URI[i].User.Username() == "" {
+			if scpOpts.URI[i].User, err = connection.GetUserInfo(scpOpts.URI[i]); err != nil {
+				return nil, err
+			}
+		}
```

Comment:
Non-blocking nit:

```suggestion
		if urlt.User.Username() == "" {
			if urlt.User, err = connection.GetUserInfo(urlt); err != nil {
				return nil, err
			}
		}
		scpOpts.URI = append(scpOpts.URI, urlt)
```
doesn’t require indexing `URI[i]`, in fact the loop index goes away.